ert_storage/endpoints/compute/misfits.py

```diff
@@ -0,0 +1,83 @@
+import numpy as np
+import pandas as pd
+from uuid import UUID
+from typing import Any, Mapping, Optional, List
+import sqlalchemy as sa
+from fastapi import (
+    APIRouter,
+    Depends,
+)
+from ert_storage.database import Session, get_db
+from ert_storage import database_schema as ds, json_schema as js
+
+router = APIRouter(tags=["misfits"])
+
+
+def _calculate_misfit(obs_value, response_value, obs_std) -> List[float]:
+    difference = response_value - obs_value
+    misfit = (difference / obs_std) ** 2
+    return (misfit * np.sign(difference)).tolist()
+
+
+@router.get(
+    "/compute/univariate_misfits",
+    response_model=Mapping[int, Mapping[str, Any]],
+)
+async def get_response_univariate_misfits(
+    *,
+    db: Session = Depends(get_db),
+    ensemble_id: UUID,
+    response_name: str,
+    realization_index: Optional[int] = None,
+) -> Mapping[int, Mapping[str, Any]]:
+    """
+    Compute univariate misfits for response(s)
+    """
+    ensemble = db.query(ds.Ensemble).filter_by(id=ensemble_id).one()
+    if realization_index is not None:
+        responses = [
+            (
+                db.query(ds.Record)
+                .filter(
+                    sa.and_(
+                        ds.Record.ensemble_pk == ensemble.pk,
+                        ds.Record.name == response_name,
+                        ds.Record.realization_index == realization_index,
+                    )
+                )
```

Comment:
Less verbose (`sa.and_` is unnecessary as it's the default when listing multiple filters):
```suggestion
                .filter_by(
                    ensemble_pk=ensemble.pk,
                    name=response_name,
                    realization_index=realization_index,
                )
```